app/services/bulk_api_import/validator.rb

```diff
@@ -21,6 +26,36 @@ def validate_schema
     {schema_errors: schema_errors} if schema_errors.present?
   end
 
+  def validate_observation_codes
+    invalid_observations = resources_by_type[:observation]&.map do |observation|
+      identifier = observation[:identifier][0][:value]
+      observation_codes = observation[:component]&.map { |component| component.dig(:code, :coding, 0, :code) }
+      next if valid_blood_pressure_codes?(observation_codes) || valid_blood_sugar_codes?(observation_codes)
+      [identifier, observation_codes]
+    end&.compact&.to_h
+
+    if invalid_observations.present?
+      {
+        invalid_observation_codes_error: {
+          invalid_observations: invalid_observations,
+          message: <<~ERRSTRING
+            Invalid set of codes for the following identifiers: #{invalid_observations.pretty_inspect}.
+            For blood pressure observation: Ensure that both systolic and diastolic measurements are present with the correct codes.
+            For blood sugar observations: Ensure that only one code is present and matching the list of accepted codes."
+          ERRSTRING
+        }
+      }
+    end
+  end
+
+  def valid_blood_pressure_codes?(codes)
+    codes.to_set == Api::V4::Imports::ALLOWED_BP_CODES.to_set
```

Comment:
Why is `codes.count == 1` not necessary here?

Reply:
A blood pressure observation has two codes associated with it: systolic and diastolic ie `Set{8480-6, 8462-4}`.
A blood sugar observation on the other hand only takes one measurement at a time, ie, either RBS or FBS/PPBS or Hb1ac.

I could have put `codes.count == 2` on this line, but that is already implicitly taken care of by the set.